api/v1alpha1/policy_types.go

```diff
@@ -60,6 +65,35 @@ func (d *Policy) IsInitialized() bool {
 }
 
 func (d *Policy) PrepareInternalValues(context context.Context, object client.Object) error {
+	// Fast path escape if no "${..}" placeholder is detected
+	match, err := regexp.MatchString("\\${[^}]+}", d.Spec.Policy)
+	if err != nil || !match {
+		return nil
+	}
+
+	log := log.FromContext(context)
+
+	// Retrieves the list of auth engines to get their accessors
+	// Kinda duplicates logic found in VaultEngineObject.retrieveAccessor
+	vaultClient := context.Value("vaultClient").(*vault.Client)
+	secret, err := vaultClient.Logical().Read("sys/auth")
+	if err != nil {
+		// Log but ignore the error: do not resolve placeholders
+		log.Error(err, "could not resolve auth engine accessor(s) in policy rule - unable to retrieve auth engines at", "path", "sys/auth")
+		return nil
+	}
+	if secret == nil {
```

Comment:
why would this result in an error if no auth are found?

Reply:
Well, the Token auth method is automatically enabled (see https://developer.hashicorp.com/vault/docs/auth/token) so no error and no auth method means something is wrong, unless I'm mistaken?